bindings/rust/bench/src/s2n_tls.rs

```diff
@@ -0,0 +1,184 @@
+use crate::{
+    harness::{Buffer, Mode, TlsImpl},
+    read_to_bytes, CA_CERT_PATH, SERVER_CERT_CHAIN_PATH, SERVER_KEY_PATH,
+};
+use log::info;
+use s2n_tls::{
+    callbacks::VerifyHostNameCallback,
+    config::{Builder, Config},
+    connection::Connection,
+    enums::Blinding,
+    security::DEFAULT_TLS13,
+};
+use std::{
+    ffi::c_void,
+    io::{Read, Write},
+    os::raw::c_int,
+    pin::Pin,
+    task::Poll::Ready,
+};
+
+pub struct S2nTls {
+    c_to_s_buf: Pin<Box<Buffer>>, // need Pin to make sure C pointer to Buffer doesn't move
```

Comment:
`Pin<Box<Buffer>>` feels like an odd team, since it's a heap allocated pointer to a heap allocated chunk of memory so is feels like there is an extra layer of indirection. We should try and get that down to the one layer of indirection.

Reply:
Nvm: I think this makes sense for now. Definitely worth adding a comment calling out why the extra layer of indirection is necessary: E.g. we need to pointer to a pointer because the bytes that the vector owns are actually jumping around in memory.